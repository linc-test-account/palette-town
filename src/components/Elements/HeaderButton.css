.header-button {
  background: var(--header-button-background);
  /* transition: .3s ease all; */
  width: 60px;
  min-width: 60px;
  height: 45px;
  overflow: hidden;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  border-style: solid;
  border-width: 0 1px 0 0;
  border-color: var(--header-button-border-color);
  text-align: center;
  cursor: pointer;
}

.header-icon {
  color: var(--app-brand-color);
  transition: .2s ease transform;
  pointer-events:none;
  font-size: 20px;
  transform: scale(1, 1);
}

.header-button:active .header-icon {
  color: rgb(255, 0, 212);
  transform: scale(.8,.8);
}


.indicator-overlay {
  font-family: var(--app-font);
  color: var(--app-text-color);
  text-shadow: 1px 1px 3px black;
  z-index: 1;
  position: absolute;
  transform: scale(1, 1);
  margin: -5px 0 0 22px;
  padding: 0;
  font-size: 14px;
  background: hsl(217, 63%, 52%);
  border-radius: 50%;
  width: 20px;
  height: 20px;
  text-align: center;
  line-height: 20px;
}
