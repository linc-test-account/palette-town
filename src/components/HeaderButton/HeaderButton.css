.button {
  background: hsl(0, 0%, 96%);
  background: var(--headerButton-background);
  border-style: solid;
  border-width: 0 0 0 1px;
  border-color: hsl(0, 0%, 88%); /* fallback */
  border-color: var(--headerButton-border-color);
  text-align: center;
  cursor: pointer;
  width: 80px;
}

.button:first-of-type {
  border-width: 0 1px 0 0;
}

.button:first-of-type {
  border-width: 0 1px 0 0;
}

.button:last-child {
  border-width: 0 1px 0 1px;
}

.icon {
  color: hsl(204, 96%, 42%);
  color: var(--app-brand-color);
  transition: 200ms cubic-bezier(.66,-0.06,.24,1.22) transform;
  pointer-events:none;
  font-size: 25px;
  transform: scale(1, 1);
}

.button:active .icon {
  color: hsl(310, 100%, 50%);
  transform: scale(.8,.8);
}

.active {
  color: hsl(310, 100%, 50%);
}

.overlay {
  font-family: var(--app-font);
  color: white;
  z-index: 1;
  position: absolute;
  transform: scale(1, 1);
  /* margin: -5px 0 0 24px; */
  padding: 0;
  font-size: 11px;
  background: hsl(217, 63%, 52%);
  border-radius: 50%;
  width: 22px;
  height: 22px;
  text-align: center;
  line-height: 22px;
  transition: 200ms ease all;
}

.button:active .overlay {
  background: hsl(310, 100%, 50%);
  transform: scale(.7,.7);
}

@media screen and (max-width: 700px) {
  .button {
    width: 50px;
  }
}
