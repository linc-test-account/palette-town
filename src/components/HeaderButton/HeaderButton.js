import React, { Component } from "react";
import { observer } from "mobx-react";
import styles from "./HeaderButton.css";
import PropTypes from "prop-types";
import classNames from "classnames";

@observer
class HeaderButton extends Component {
  static propTypes = {
    btnFunction: PropTypes.func,
    fontAwesomeIcon: PropTypes.string,
    overlayValue: PropTypes.number,
    isActive: PropTypes.bool
  };
  render() {
    const { btnFunction, fontAwesomeIcon, overlayValue, isActive } = this.props;
    return (
      <button className={styles.button} onClick={btnFunction}>
        <i
          className={classNames({
            [fontAwesomeIcon]: true,
            [styles.icon]: true,
            [styles.active]: isActive
          })}
        />
        {overlayValue > 0 ? (
          <p className={styles.overlay}>{overlayValue}</p>
        ) : (
          ""
        )}
      </button>
    );
  }
}

export default HeaderButton;
