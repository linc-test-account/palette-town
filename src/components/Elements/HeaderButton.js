import React, { Component } from "react";
import { observer } from "mobx-react";
import FontAwesome from "react-fontawesome";
import "./HeaderButton.css";
import PropTypes from "prop-types";

@observer
class HeaderButton extends Component {
  static propTypes = {
    btnFunction: PropTypes.func,
    fontAwesomeIcon: PropTypes.string,
    overlayValue: PropTypes.number,
  };
  render() {
    const { btnFunction, fontAwesomeIcon, overlayValue } = this.props;
    return (
      <button className="header-button" onClick={btnFunction}>
        <FontAwesome
          className="header-icon"
          name={fontAwesomeIcon}
          size="2x"
        />
        {overlayValue > 0 ? (
          <p className="indicator-overlay">{overlayValue}</p>
        ) : (
          ""
        )}
      </button>
    );
  }
}

export default HeaderButton;
