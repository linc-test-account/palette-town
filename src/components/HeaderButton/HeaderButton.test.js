import HeaderButton from "./HeaderButton";

import React from "react";
import { shallow } from "enzyme";

describe("HeaderButton", () => {
  let props;
  beforeEach(() => {
    props = {
      btnFunction: jest.fn(),
      fontAwesomeIcon: "exampleIcon",
      overlayValue: 1,
      isActive: true
    };
  });
  
  it("should render active", () => {
    const wrapper = shallow(<HeaderButton {...props} isActive={true} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should render in-active", () => {
    const wrapper = shallow(<HeaderButton {...props} isActive={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should call when clicked", () => {

    const wrapper = shallow(<HeaderButton {...props} isActive={false} />);
    
    wrapper.find('button').simulate('click');

    expect(props.btnFunction).toHaveBeenCalledTimes(1);
  })
});
