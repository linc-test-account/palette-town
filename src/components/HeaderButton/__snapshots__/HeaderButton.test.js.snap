// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton should render active 1`] = `
<button
  onClick={[Function]}
>
  <FontAwesome
    name="exampleIcon"
    size="2x"
    style={
      Object {
        "color": "hsl(310, 100%, 50%)",
      }
    }
  />
  <p>
    1
  </p>
</button>
`;

exports[`HeaderButton should render in-active 1`] = `
<button
  onClick={[Function]}
>
  <FontAwesome
    name="exampleIcon"
    size="2x"
    style={
      Object {
        "color": "",
      }
    }
  />
  <p>
    1
  </p>
</button>
`;
